### PR TITLE
feat(MenuInput): use SearchInput instead of TextInput

### LINKED ISSUE #7213: [Menu] - [Replace menuInput with search input]
**Is this a new component or an extension of an existing one?**
What is the existing component, if any?
Menu

**Describe the feature**
A clear and concise description of the new feature. What is the expected behavior?

Update examples and demos that use `MenuInput` to use  `SearchInput instead of `TextInput`

### PR DESCRIPTION
**Pull request by tompsota**
**What**: Closes #7213

### CODE REVIEW

**Review by thatblindgeye — COMMENTED** [review]
This is looking good! There's just some more tweaking that needs to be done in the Menu integration test (the tabshorizontaloverflow should be safe to ignore, it should pass when it's re-ran)

**Review by thatblindgeye — APPROVED** [review]

File: packages/react-core/src/demos/ComposableMenu/examples/ComposableApplicationLauncher.tsx [hunk 1]
```diff
@@ -250,12 +250,7 @@ export const ComposableApplicationLauncher: React.FunctionComponent = () => {
     // eslint-disable-next-line no-console
     <Menu ref={menuRef} onActionClick={onFavorite} onSelect={(_ev, itemId) => console.log('selected', itemId)}>
       <MenuInput>
-        <TextInput
-          aria-label="Filter menu items"
-          iconVariant="search"
-          type="search"
-          onChange={value => onTextChange(value)}
-        />
+        <SearchInput aria-label="Filter menu items" type="search" onChange={value => onTextChange(value)} />
```
Comment (gitdallas) [inline packages/react-core/src/demos/ComposableMenu/examples/ComposableApplicationLauncher.tsx:253]: @tlabaj - any reason SearchInput type isn't defaulted to search?

**Review by gitdallas — APPROVED** [review]
lgtm, just a question for @tlabaj below

**Review by mattnolting — APPROVED** [review]
Looks great! 👍

**Review by mmenestr — APPROVED** [review]
LGTM!

**Review by kmcfaul — APPROVED** [review]

File: packages/react-core/src/demos/ComposableMenu/examples/ComposableApplicationLauncher.tsx (hunk 1 above)
Comment (kmcfaul) [reply packages/react-core/src/demos/ComposableMenu/examples/ComposableApplicationLauncher.tsx:253]: It looks like when I added it in that I matched the default `type` of TextInputGroupMain which was `text`. It probably could be defaulted to `search` instead though.

**Review by tlabaj — APPROVED** [review]
lgtm

File: packages/react-core/src/demos/ComposableMenu/examples/ComposableApplicationLauncher.tsx (hunk 1 above)
Comment (thatblindgeye) [reply packages/react-core/src/demos/ComposableMenu/examples/ComposableApplicationLauncher.tsx:253]: @tlabaj I actually have the following PR open in Core that relates to the "x" icon for search input types 
https://github.com/patternfly/patternfly/pull/5211

**Pull request merged**